### PR TITLE
Fixes #13150: Add documentation to the Vault plugin

### PR DESCRIPTION
**Pull request by VictorQrt**
https://www.rudder-project.org/redmine/issues/13150

### CODE REVIEW

File: vault/README.md [hunk 1]
```diff
@@ -9,12 +9,39 @@ Mailing list, irc : https://www.rudder-project.org/site/community/mailing-lists/
...
+  ```
+
+- Edit the configuration file in /var/rudder/plugin-resources/vault.json on each agent. This config file must contain
+  the address of your Vault server, credentials to access it and the auth mode you want to use. A sample config is in
+  share/plugins/vault/sample_vault.json.
+
 Usage
 -----
 
+Use the Variable_from_Vault generic method in Rudder to fetch secrets. Make sure the agents the generic method is being
```
Comment (peckpeck) [inline vault/README.md]: lowercase Variable_from_Vault

File: vault/README.md [hunk 2]
```diff
@@ -9,12 +9,39 @@ Mailing list, irc : https://www.rudder-project.org/site/community/mailing-lists/
...
+used on have a proper vault.json configuration.
```
Comment (peckpeck) [inline vault/README.md]: And the documentation for this file ?

**Comment by VictorQrt** [comment]
PR rebased

**Comment by VictorQrt** [comment]
Commit modified

**Pull request merged**